lib/IDE/Utils.cpp

```diff
@@ -649,15 +649,22 @@ accept(SourceManager &SM, RegionType Type, ArrayRef<Replacement> Replacements) {
 void swift::ide::SourceEditTextConsumer::
 accept(SourceManager &SM, RegionType Type, ArrayRef<Replacement> Replacements) {
   for (const auto &Replacement: Replacements) {
-    CharSourceRange Range = Replacement.Range;
-    unsigned BufID = SM.findBufferContainingLoc(Range.getStart());
-    auto Path(SM.getIdentifierForBuffer(BufID));
-    auto Start = SM.getLineAndColumnInBuffer(Range.getStart());
-    auto End = SM.getLineAndColumnInBuffer(Range.getEnd());
+    OS << "// ";
+    StringRef Path = Replacement.Path;
+    if (Path.empty()) {
+      unsigned BufID = SM.findBufferContainingLoc(Replacement.Range.getStart());
+      Path = SM.getIdentifierForBuffer(BufID);
+    } else {
+      OS << "explicit ";
+    }
+    OS << Path.str() << " ";
 
-    OS << "// " << Path.str() << " ";
+    auto Start = SM.getLineAndColumnInBuffer(Replacement.Range.getStart());
+    auto End = SM.getLineAndColumnInBuffer(Replacement.Range.getEnd());
     OS << Start.first << ":" << Start.second << " -> ";
-    OS << End.first << ":" << End.second << "\n";
+    OS << End.first << ":" << End.second << " (";
+
+    OS << Replacement.BufferName << ")\n";
```

Comment:
Do we want to print this unconditionally?

Reply:
Ah thanks, probably not. Though it shouldn't *really* matter, but not having trailing () makes sense :)